src/usePersistStorage.ts

```diff
@@ -148,7 +153,15 @@ const usePersistStorage = <Value>(
     }
   };
 
-  return [state, asyncSetState, restored];
+  const resetState = async () => {
+    const newValue: Value =
+      initialValue instanceof Function
+        ? initialValue()
+        : initialValue;
+    await asyncSetState(newValue);
```

Comment:
It would be better if we can wrap `useCallback`, and reduce the dependency.
I will suggest you re-write to  like below...
```
const resetState = useCallback(async () => {
    const newValue: Value =
      initialValue instanceof Function
        ? initialValue()
        : initialValue;
  
    setState(newValue);
    if (persist) {
      await setValueToStorage(newValue);
    }
  }, [initialValue, persist, setValueToStorage]);
```